### ISSUE TITLE
Rationalise the GitHub and Buildkite permissions

### ISSUE THREAD

**Issue by alexwlchan** [wellcomecollection/platform#5620]

**Comment by alexwlchan** [comment]
I've rationalised the list of teams:

* `Buildkite reviewers` is gone; we trust all devs to manage Buildkite permissions. There was an idea at one point that all Buildkite changes would go through a select group of reviewers, but in practice that was more a hindrance than helpful.
* I've moved `Identity` to be a subteam of `Developers` to simplify the top-level view.

I've also reviewed all read-only permissions on repos:

Archives I’m going to leave as is.

* `archivematica` = I’ve given all devs write access, even though most of them should be going away soon.
* `archivematica-docs` = deleted. It was a fork we created for the OIDC work which is now merged upstream.
* `aws_utils` = archived, we haven’t touched it in ages
* `identity` = I’ve given all devs write access, because all JS/TS reviewers have write access and in practice that’s most of the team
* `prismic-remapper` `python-gearman` `tzitzit` = all devs get write access
* `wellcomeplayer*` – tbh these should probably all be public archives

This is all the low-hanging fruit, but happy to use my admin powers to make other changes people suggest – I'm not sure what else to do.
